### PR TITLE
Update Devise from 3.5.2 to 4.6.1

### PR DESCRIPTION
**Pull request by baphled**
Resolves CVE-2019-5421

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -19,9 +19,9 @@ gem "sdoc", "~> 0.4.0", group: :doc
 gem "pg", "~> 0.20.0"
 gem "dotenv-rails"
 
-gem "devise",           "~> 3.5.2"
-gem "devise_invitable", "~> 1.5.2"
-gem "devise_security_extension"
+gem "devise",           "~> 4.6.1"
+gem "devise_invitable", "~> 1.7.5"
+gem 'devise_security_extension', github: 'phatworx/devise_security_extension'
```
Comment (MattHallEA) [inline Gemfile]: Why are we switching to use an unreleased version here?

Comment (baphled) [reply Gemfile]: The current release removed `password_salt` from `OldPassword`. This branch resolves that.

Comment (baphled) [reply Gemfile]: Happy to use the main repo, once we've unblocked, but for now I decided to use this branch to limit the amount of change going on at the moment.

Comment (MattHallEA) [reply Gemfile]: We need to move to a maintained version of this gem: https://github.com/devise-security/devise-security - Adding a ticket for future work to fix this up.

**Comment by MattHallEA** [comment]
This PR will depend on https://github.com/DEFRA/pafs_core/pull/234

**Pull request merged**